### PR TITLE
ref(ui): Refactor DropdownAutoCompleteMenu to allow render func for menu header

### PR DESCRIPTION
**Pull request by billyvg**
This allows menu header to be a render function and refactors so that
 render functions can override padding.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx [hunk 1]
```diff
@@ -18,6 +18,24 @@ const ItemObjectPropType = {
 };
 const ItemShapePropType = PropTypes.shape(ItemObjectPropType);
 
+// Use before define :(
```
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: 😭

File: src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx [hunk 2]
```diff
@@ -317,8 +335,15 @@ class DropdownAutoCompleteMenu extends React.Component {
           // emptyHidesInput is set to true.
           let showInput = hasItems || !emptyHidesInput;
 
+          let renderProps = {actions, components: {LabelWithPadding}};
```
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: is it worth it to do this, or should we just extract this stuff to a new component?

**Review by Chrissy — APPROVED** [review]
this is kinda wild but I like it

File: src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: Yeah this might just be better off as an importable component.

**Comment by BYK** [comment]
I don't think this is relevant anymore, so closing.

**Pull request closed without merging**